include/swift/SIL/SILFunction.h

```diff
@@ -123,6 +123,9 @@ class SILFunction
   /// The forwarding substitutions, lazily computed.
   Optional<SubstitutionList> ForwardingSubs;
 
+  /// The forwarding substitution map, lazily computed.
+  SubstitutionMap ForwardingSubMap;
```

Comment:
Perhaps we should move this caching logic to GenericEnvironment?

Reply:
So little code cares about it that it doesn't seem necessary.